### ISSUE TITLE
Postman eats a lot of CPU and kills it within minutes

### ISSUE THREAD

**Issue by jayesh15111988** [a85/POSTMan-Chrome-Extension#11976]
### Is there an existing issue for this?

- [X] I have searched the tracker for existing similar issues and I know that duplicates will be closed

### Describe the Issue

Every time I open the Postman app on Mac, my Mac starts making sound as it doing an intensive computation. I am using the latest Postman. My preference has always been using the web client, but it forces me to use the desktop client for uploading file (Which should not be required).

For now, I am trying not to use it as much as possible, but I wonder what makes this seemingly small app CPU hungry within minutes.

### Steps To Reproduce

1. Just open Postman Mac client app

### Operating System

MacOS

### Postman Version

10.13.5

### Postman Platform

Postman App

### User Account Type

Signed In User

### Additional Context?

Nothing else to add

**Comment by savedario** [comment]
I am having a similar issue. After starting the Postman (v10.14.3 but the previous version was doing the same since a recent update ~12/05/2023), I see 3 'postman' processes using CPU.
2 slowly disappear but one remains using > 130% CPU even though it's doing nothing but drawing the UI.
A minute later the fan start spinning (that's how I discovered it).
When started from CLI, I see this at the start of the output:
`The disableGPU setting is set to undefined
Not disabling GPU
[19472:0518/180624.077216:ERROR:sandbox_linux.cc(377)] InitializeSandbox() called with multiple threads in process gpu-process.
`
but the rest is only 'info'.
I deleted my install and downloaded and re-installed. The issue remains.

(Running OpenSUSE Tumbleweed 20230517, KDE Plasma 5.27.5, 32GB RAM

**Comment by jviotti** [comment]
@savedario What is the name of the third process using > 130% CPU? Can you post a screenshot?

**Comment by sandystorm1990** [comment]
<img width="928" alt="image" src="https://github.com/postmanlabs/postman-app-support/assets/15962932/3116aff9-012b-48fe-98a3-299d5e7cb9b0">

**Comment by zengzexiong** [comment]
![image](https://github.com/postmanlabs/postman-app-support/assets/27144161/8da3efb3-a413-4011-a0c4-dcdb58f02281)
![image](https://github.com/postmanlabs/postman-app-support/assets/27144161/02cdf4c5-d360-43f2-87ff-96ec4bb53eb7)
The operating system is a windows environment, postman is started, and no requests are run. CPU usage is very high.
Updated to latest version, but no change. The current version is v10.17.4

**Comment by jayesh15111988** [comment]
Looks like no one really knows what's going on. I guess this is a feature, not a bug! Great job! Closing after 4 months of creating issue with no resolution in mind.

**Comment by kamalaknn** [comment]
We've fixed this in 10.18. This is being rolled out now. Do update your apps and try it let. Let us know if you are still facing the issue.

**Comment by snowman** [comment]
![image](https://github.com/postmanlabs/postman-app-support/assets/16096559/066bf6a1-c70e-4635-9ea5-1439894b86c0)
![33331](https://github.com/postmanlabs/postman-app-support/assets/16096559/1120889d-68c0-4fb9-b1a0-437622b6041c)

**Comment by piotreeeek** [comment]
I'm just installed it, open and immediately fans spinning. Great app, great success, great devs.

**Comment by panamantis** [comment]
Fresh install Dec 2023, windows 64.  Ultra high cpu while idle in background.  Used for 15 minutes.  If it's been happening for multiple years then likely mining coins.

**Comment by beryozavv** [comment]
> Fresh install Dec 2023, windows 64. Ultra high cpu while idle in background. Used for 15 minutes. If it's been happening for multiple years then likely mining coins.

Same Here. The Postman processes consume about 30% of cpu when the Postman in idle after several requests

**Comment by attikov** [comment]
It's burning my cpu, killing my notebook, consumes energy, Greta is already out of her mind, please hire an expert.

**Comment by JorgeSivil** [comment]
It's been happening since I've been using Postman (except early versions).

I don't understand how it can get my processors to 140% without Postman being used. It's just there, open, then your computer starts being very slow, then you notice it's postman.

I'm in Ubuntu.

Is it possible to be related to low memory? I closed all the apps and opened Postman, and no more CPU issues.
I had 1.5gb free

**Comment by danekszy** [comment]
Hey @JorgeSivil, mind sharing any more info about your setup?
Which Postman version are you using?
Are you working with any large data-sets (lots of requests, big request response etc)?
Regarding your question about memory usage: Do you mind checking how much memory is Postman using at the time that you're experiencing the issue? This would be hugely helpful for us to triage what might be happening here! 😉 
Thanks for pinging us with this! 🙌

**Comment by attikov** [comment]
Last time I encountered this issue, I didn't exit Postman; instead, I terminated one of the Postman processes that was using over 600MB of memory and consistently consuming about 14% of CPU resources (equivalent to one core's usage).
Killing this process had no impact on Postman's functionality at all. This was on Windows 10 while I was testing the DHL Rest API, which is pretty basic and involves very little data, including base64 encoded labels.

**Comment by danekszy** [comment]
@attikov Sounds like it could be Service Worker responsible for data synchronization... Are you logged in? Can you check if the changes saved in the app appear on the web version of Postman?

**Comment by attikov** [comment]
Currently, it is working. I'll check it again next time. By the way, I was not aware that entering sensitive data into Postman could result in it being synced to an unspecified server. I'm not comfortable with this.

**Comment by kamalaknn** [comment]
@JorgeSivil I want to rule out an issue we had seen [earlier](https://github.com/postmanlabs/postman-app-support/issues/7294#issuecomment-1669001260).

Could you follow the steps and let us know if you find anything?

1.  Go to `View -> Developer -> View logs in Finder`
2. Go to the parent folder
3. Go to `storage` folder
4. Look for any files ending with `.lock`

**Comment by JorgeSivil** [comment]
@kamalaknn no lock files

```
jorgee@jorgee:~/snap/postman/233/.config/Postman/logs$ ls -la
total 19092
drwxrwxr-x  2 jorgee jorgee    4096 ene 17 16:10 .
drwx------ 13 jorgee jorgee    4096 feb 26 12:55 ..
-rw-rw-r--  1 jorgee jorgee   71992 feb 26 12:55 main.log
-rw-rw-r--  1 jorgee jorgee     305 abr  3  2023 renderer-auth.log
-rw-rw-r--  1 jorgee jorgee 5244083 oct 13 13:30 renderer-requester1.log
-rw-rw-r--  1 jorgee jorgee 5245149 ene 11 15:29 renderer-requester2.log
-rw-rw-r--  1 jorgee jorgee 3349744 feb 26 13:52 renderer-requester3.log
-rw-rw-r--  1 jorgee jorgee  176409 ene 17 10:29 renderer-requester4.log
-rw-rw-r--  1 jorgee jorgee 5246257 sep  6 12:50 renderer-requester.log
-rw-rw-r--  1 jorgee jorgee  170419 feb 26 12:55 renderer-shell.log
```

bwt I have less than 50 requests

[main.log](https://github.com/postmanlabs/postman-app-support/files/14408368/main.log)
[renderer-requester3.log](https://github.com/postmanlabs/postman-app-support/files/14408369/renderer-requester3.log)
[renderer-shell.log](https://github.com/postmanlabs/postman-app-support/files/14408371/renderer-shell.log)


What info do you need next time it happens

**Comment by kamalaknn** [comment]
@JorgeSivil Thanks for getting back with the logs.

The lock files if present should be in `~/snap/postman/233/.config/Postman/storage`. The details you have shared are for the `logs` directory. Could you check the storage directory?

Also it seems that you are on version `9.31.30`. We fixed one of the high CPU issues in a v10 version. Could you update your app to the latest version and let us know if this is still an issue?

**Comment by JorgeSivil** [comment]
> @JorgeSivil Thanks for getting back with the logs.
> 
> The lock files if present should be in `~/snap/postman/233/.config/Postman/storage`. The details you have shared are for the `logs` directory. Could you check the storage directory?
> 
> Also it seems that you are on version `9.31.30`. We fixed one of the high CPU issues in a v10 version. Could you update your app to the latest version and let us know if this is still an issue?

```
jorgee@jorgee:~$ ls -la  ~/snap/postman/233/.config/Postman/storage
total 20
drwxrwxr-x  2 jorgee jorgee 4096 feb 26 12:55 .
drwx------ 13 jorgee jorgee 4096 feb 26 12:55 ..
-rw-rw-r--  1 jorgee jorgee   39 feb 26 12:55 requester.json
-rw-rw-r--  1 jorgee jorgee   58 feb 26 12:55 settings.json
-rw-rw-r--  1 jorgee jorgee  546 feb 26 12:55 userPartitionData.json

```

**Comment by hwebb** [comment]
Same thing here, fresh install from today from the download on the website (v10.23).
12-20% of CPU (i9-12900k) without anything running! Fortunately I'm not using this tool frequently enough to have a meaningful impact, I remember it was doing the same last year on another setup.

**Comment by simeyla** [comment]
I'm on Windows and haven't done much investigation, but my fans were out of control for hours and it was using 30% CPU with nothing happening - even when minimized.

On closing and reopening I got a message about Windows Firewall having blocked Postman. I rarely use it - let alone have it open for an entire day, so I don't know if this is a new issue or not for me. I didn't check for a .lock file while it was slow.

If it is as simple as not having network access and it keeps retying then that seems like a trivial fix that shouldn't have ever been an issue in the first place. I'll edit this if the problem returns.

**Comment by shreyakqss** [comment]
I just ended up using the web version of Postman with desktop agent. Having 30% CPU usage in idle on Windows x64

**Comment by kamalaknn** [comment]
@shreyakqss Could you help us with some additional information to identify the cause for the issues?

1. Could you share the application logs with us at help@postman.com? If you are able to open the app you can find them under `View -> Developer -> View logs in Explorer`. 
2. Could you share the list of files present in the storage directory, it should be in `C:\Users\<user>\AppData\Local\Postman\storage` or 
`C:\Users\<user>\AppData\Roaming\Postman\storage`.

**Comment by wileyveteran** [comment]
My postman app (on Windows) was using 25-30% of CPU so I checked the C:\Users\<user>\AppData\Roaming\Postman\storage folder and had a settings.json.lock file and another for userPartitionData.json that was suffixed with STALE. I shutdown Postman to make sure nothing was running, deleted both of those files and restarted postman and Postman CPU is now back to 0 when not using it.

**Comment by marius-oloeru** [comment]
In my case disabling hardware acceleration fixed the problem.

**Comment by ZakariaX1** [comment]
> In my case disabling hardware acceleration fixed the problem.

This seems to be the best solution so far, I would love to know what this hardware acceleration is doing for the app 🤔 
I don't notice any difference in the rendering of the app/visuals, so I'd love to know what it's doing. 
Thanks @marius-oloeru !

**Comment by armandomejiam** [comment]
TL;DR disabling hardware acceleration works for me!

I uninstalled Postman months ago because of this issue, the laptop fan was always noisy when the Postman app was open, and then when I checked the resources on the Task Manager, the CPU consumption was always too high.

The thing is I need Postman, and the VS Code extension was not suitable for me, nor the web client...

So I installed the app again today and the CPU went high usage again in matter of minutes, so I disabled the hardware acceleration and this time it seems to be behaving decently.

Thanks @marius-oloeru

**Comment by snowman** [comment]
To disable `Hardware acceleration`, [Troubleshoot app issues - Postman Learning Center](https://learning.postman.com/docs/introduction/troubleshooting-inapp/#hardware-acceleration)

> Hardware acceleration
>
> Postman takes advantage of your computer's built-in graphics hardware
> to speed up the rendering of onscreen graphics.
>
> Sometimes your computer's specific GPU, drivers, or operating system
> may cause issues with hardware acceleration.
>
> If you see screen artifacts or other glitches, turning off hardware
> acceleration might solve the problem.
>
> To activate or deactivate hardware acceleration in Postman:
>
>     On macOS, select Postman > Hardware Acceleration.
>     On Windows or Linux, select Help > Hardware Acceleration.
>
> Restart Postman to apply the new setting.
> Disabling hardware acceleration may affect performance or CPU usage.

**Comment by snowman** [comment]
To disable `Hardware acceleration`, [Troubleshoot app issues - Postman Learning Center](https://learning.postman.com/docs/introduction/troubleshooting-inapp/#hardware-acceleration)

> Hardware acceleration
>
> Postman takes advantage of your computer's built-in graphics hardware
> to speed up the rendering of onscreen graphics.
>
> Sometimes your computer's specific GPU, drivers, or operating system
> may cause issues with hardware acceleration.
>
> If you see screen artifacts or other glitches, turning off hardware
> acceleration might solve the problem.
>
> To activate or deactivate hardware acceleration in Postman:
>
>     On macOS,            select Postman > Hardware Acceleration
>     On Windows or Linux, select Help    > Hardware Acceleration
>
> Restart Postman to apply the new setting.
> Disabling hardware acceleration may affect performance or CPU usage.

**Comment by snowman** [comment]
To disable `Hardware acceleration`, read [Troubleshoot app issues - Postman Learning Center](https://learning.postman.com/docs/introduction/troubleshooting-inapp/#hardware-acceleration)

> Hardware acceleration
>
> Postman takes advantage of your computer's built-in graphics hardware to speed up the rendering of onscreen graphics.
>
> Sometimes your computer's specific GPU, drivers, or operating system may cause issues with hardware acceleration.
>
> If you see screen artifacts or other glitches, turning off hardware acceleration might solve the problem.
>
> To activate or deactivate hardware acceleration in Postman:
>
>     On macOS,            select Postman > Hardware Acceleration
>     On Windows or Linux, select Help    > Hardware Acceleration
>
> Restart Postman to apply the new setting.
> Disabling hardware acceleration may affect performance or CPU usage.

**Comment by Vignatus** [comment]
@marius-oloeru Thanks
Disabling hardware acceleration for version 11.1.25 reduced the memory footprint from above 2 GB to ~260 MB on Mac M1. More like Hardware Deceleration.

Unless this issue is fixed, future updates should keep it off as default

**Comment by madzohan** [comment]
lol i thought it has integrated crypto miner 😄 it's not fixed YET 😞 on **11.5.1-ui-240724-0915**

**Comment by lantanios** [comment]
> lol i thought it has integrated crypto miner 😄 it's not fixed YET 😞 on **11.5.1-ui-240724-0915**

Having this issue on Version 11.17.1

Has anyone inspected traffic with Wireshark or analyzed what is actually going on?

It can be a backdoor into computer.

Why it's so hard to resolve?

**Comment by numaanashraf** [comment]
@lantanios Could you try [disabling](https://learning.postman.com/docs/introduction/troubleshooting-inapp/#hardware-acceleration) 'hardware acceleration'?

**Comment by yacoll** [comment]
I was struggling with Postman consuming a lot of my CPU for months... CPU fan spinning like creazy :(
Today I found this conversation and I disabled "hardware acceleration". WHAT A RELIEF!!! Pure silence...
Thank you Guys. It works!

**Comment by lantanios** [comment]
It's doing that randomly, just restarting the app stopped it and it didn't start again for days, then it happened again. I stopped using the app, until will hear the issue was identified and fixed.

Disabling "hardware acceleration" is not a solution, if someone has integrated a trojan miner or whatever it is, it needs to be found and identified by the dev team.

**Comment by illiaboiko** [comment]
for a record. I installed Postman windows client on Dec 28, 2024 ver.11.23.3.  win 10
I made very simple requests to a local server. Fans and CPU usage was quite an issue. Restarted the app and now it's working fine.
Who knows maybe it'll happen again in future

**Comment by oscar-o-oneill** [comment]
Just adding my experience of running postman desktop app version 11.28.4, UI version 11.28.4-ui-250116-2337 on Windows 11 and the app started using a ton of CPU (40-50%) even when the app was idle. I disabled hardware acceleration and it's not using as much CPU now, only 1% at idle.

**Comment by romanberdnikov** [comment]
The app was closed, however it remained in process list. After I left, the PC started to work noisily. It turned out that postman was taking about 50% CPU with about 10Mb of RAM.
Hidden miner?

**Comment by mfrieling** [comment]
I have the same issue with version 11.34.x on Windows 11 now. Just starting Postman and the fan starts spinning up after a few seconds like you do an engine test on an aircraft.  Just updated to 11.35.0 but it stays the same. CPU usage of Postman varies between 0.5% and 45%, but interestingly Chrome CPU usage increases by a few percent and Chrome Memory usage by 400MB at the same time while the Chrome window is idle.

**Comment by LucidObscurity** [comment]
I found disabling "send anonymous usage data to postman" fixed the issue right away.

**Comment by prplecake** [comment]
Looks like this is still an issue in 11.44! Disabling hardware acceleration helped. It should be off by default if it's causing all these problems.

**Comment by Aedon** [comment]
So I cleared cache and restarted (menu->help) which brought it down from ~60% to ~7. I then followed what @LucidObscurity said and then I was looking at <1%
